<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width">
<meta name="theme-color" content="#222"><meta name="generator" content="Hexo 6.3.0">

  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/cucumber_32.ico">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/cucumber_16.ico">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css" integrity="sha256-HtsXJanqjKTc8vVQjO4YMhiqFoXkfBsjBWcX91T1jr8=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.1.1/animate.min.css" integrity="sha256-PR7ttpcvz8qrF57fur/yAx1qXMFJeJFiA6pSzWi0OIE=" crossorigin="anonymous">

<script class="next-config" data-name="main" type="application/json">{"hostname":"yetu.fun","root":"/","images":"/images","scheme":"Gemini","darkmode":false,"version":"8.16.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":{"enable":"ture","style":"mac"},"bookmark":{"enable":false,"color":"#222","save":"auto"},"mediumzoom":false,"lazyload":true,"pangu":true,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"stickytabs":false,"motion":{"enable":true,"async":false,"transition":{"menu_item":"fadeInDown","post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"搜索...","empty":"没有找到任何搜索结果：${query}","hits_time":"找到 ${hits} 个搜索结果（用时 ${time} 毫秒）","hits":"找到 ${hits} 个搜索结果"},"path":"/search.json","localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false}}</script><script src="/js/config.js"></script>

    <meta name="description" content="转载于网络，原文地址 www.plob.org  一、基础知识1. 解密表观遗传学的三个方向与测序方法">
<meta property="og:type" content="article">
<meta property="og:title" content="ATAC分析流程">
<meta property="og:url" content="https://yetu.fun/post/786e1faa.html">
<meta property="og:site_name" content="Yetu">
<meta property="og:description" content="转载于网络，原文地址 www.plob.org  一、基础知识1. 解密表观遗传学的三个方向与测序方法">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632949992-700x506.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632949994-700x844.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632949997-700x318.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950000-700x238.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950000-1.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950001-700x514.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950001-1-700x296.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950002-700x344.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950005-700x773.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950006-700x683.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950007.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950007-1-700x503.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950008-700x407.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950009-700x436.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950010-700x629.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950011-700x630.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950013-700x649.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950014-700x605.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950015-700x734.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950016-700x323.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950019-700x220.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950021-700x685.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950022-700x342.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950023-700x261.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950025-700x148.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950026-700x127.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950027.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950027-1-700x349.png">
<meta property="og:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632950029-700x643.png">
<meta property="article:published_time" content="2023-05-18T17:39:47.000Z">
<meta property="article:modified_time" content="2023-05-18T17:39:47.000Z">
<meta property="article:author" content="Huang">
<meta property="article:tag" content="note">
<meta property="article:tag" content="ATAC">
<meta property="article:tag" content="pipeline">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://static.plob.org/wp-content/uploads/2021/09/1632949992-700x506.png">


<link rel="canonical" href="https://yetu.fun/post/786e1faa.html">



<script class="next-config" data-name="page" type="application/json">{"sidebar":"","isHome":false,"isPost":true,"lang":"zh-CN","comments":true,"permalink":"https://yetu.fun/post/786e1faa.html","path":"post/786e1faa.html","title":"ATAC分析流程"}</script>

<script class="next-config" data-name="calendar" type="application/json">""</script>
<title>ATAC分析流程 | Yetu</title>
  

  <script src="/js/third-party/analytics/baidu-analytics.js"></script>
  <script async src="https://hm.baidu.com/hm.js?b43fc0154a1c371255b3e8e08dc92be8"></script>







  <noscript>
    <link rel="stylesheet" href="/css/noscript.css">
  </noscript>
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.css" integrity="sha384-bsHo4/LA+lkZv61JspMDQB9QP1TtO4IgOf2yYS+J6VdAYLVyx1c3XKcsHh0Vy8Ws" crossorigin="anonymous">
  <script defer src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.js" integrity="sha384-4z8mjH4yIpuK9dIQGR1JwbrfYsStrNK6MP+2Enhue4eyo0XlBDXOIPc8b6ZU0ajz" crossorigin="anonymous">
  </script>
</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>
  <!--
  <a target="_blank" rel="noopener" href="https://github.com/WhaleGe" class="github-corner" aria-label="View source on GitHub"><svg width="80" height="80" viewBox="0 0 250 250" style="fill:#151513; color:#fff; position: absolute; top: 0; border: 0; right: 0;" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a><style>.github-corner:hover .octo-arm{animation:octocat-wave 560ms ease-in-out}@keyframes octocat-wave{0%,100%{transform:rotate(0)}20%,60%{transform:rotate(-25deg)}40%,80%{transform:rotate(10deg)}}@media (max-width:500px){.github-corner:hover .octo-arm{animation:none}.github-corner .octo-arm{animation:octocat-wave 560ms ease-in-out}}</style>
  -->

  <main class="main">
    <div class="column">
      <header class="header" itemscope itemtype="http://schema.org/WPHeader"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="切换导航栏" role="button">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <p class="site-title">Yetu</p>
      <i class="logo-line"></i>
    </a>
      <p class="site-subtitle" itemprop="description">Blog</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger" aria-label="搜索" role="button">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu"><li class="menu-item menu-item-home"><a href="/home/" rel="section"><i class="fa fa-home fa-fw"></i>首页</a></li><li class="menu-item menu-item-categories"><a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>分类</a></li><li class="menu-item menu-item-archives"><a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>归档</a></li><li class="menu-item menu-item-tags"><a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>标签</a></li><li class="menu-item menu-item-about"><a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>关于</a></li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>搜索
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup"><div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off" maxlength="80"
           placeholder="搜索..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close" role="button">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div class="search-result-container no-result">
  <div class="search-result-icon">
    <i class="fa fa-spinner fa-pulse fa-5x"></i>
  </div>
</div>

    </div>
  </div>

</header>
        
  
  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          文章目录
        </li>
        <li class="sidebar-nav-overview">
          站点概览
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E4%B8%80%E3%80%81%E5%9F%BA%E7%A1%80%E7%9F%A5%E8%AF%86"><span class="nav-text">一、基础知识</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#1-%E8%A7%A3%E5%AF%86%E8%A1%A8%E8%A7%82%E9%81%97%E4%BC%A0%E5%AD%A6%E7%9A%84%E4%B8%89%E4%B8%AA%E6%96%B9%E5%90%91%E4%B8%8E%E6%B5%8B%E5%BA%8F%E6%96%B9%E6%B3%95"><span class="nav-text">1. 解密表观遗传学的三个方向与测序方法</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#2-%E5%90%8D%E8%AF%8D%E8%A7%A3%E9%87%8A"><span class="nav-text">2. 名词解释</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E4%BA%8C%E3%80%81%E5%9F%BA%E7%A1%80%E5%88%86%E6%9E%90"><span class="nav-text">二、基础分析</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#1-%E5%88%86%E6%9E%90%E5%89%8D%E7%9A%84%E5%87%86%E5%A4%87"><span class="nav-text">1. 分析前的准备</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-%E6%B5%8B%E5%BA%8F%E8%B4%A8%E9%87%8F%E6%A3%80%E6%B5%8B"><span class="nav-text">2. 测序质量检测</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-%E6%B5%8B%E5%BA%8F%E8%B4%A8%E9%87%8F%E6%A3%80%E6%8E%A7%E5%88%B6"><span class="nav-text">2. 测序质量检控制</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-%E5%BB%BA%E7%AB%8B%E7%B4%A2%E5%BC%95%E4%B8%8E%E5%BA%8F%E5%88%97%E6%AF%94%E5%AF%B9"><span class="nav-text">3. 建立索引与序列比对</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-%E5%8E%BB%E9%99%A4-PCR-%E9%87%8D%E5%A4%8D%E5%B9%B6%E5%86%8D%E6%AC%A1%E8%BF%9B%E8%A1%8C%E8%B4%A8%E6%8E%A7"><span class="nav-text">4. 去除 PCR 重复并再次进行质控</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#5-%E7%BB%9F%E8%AE%A1%E8%BF%87%E6%BB%A4%E5%90%8E%E7%9A%84%E7%BB%93%E6%9E%9C%E4%B8%AD-Insertsize-%E9%95%BF%E5%BA%A6%E5%88%86%E5%B8%83"><span class="nav-text">5. 统计过滤后的结果中 Insertsize 长度分布</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#5-peak-calling-%E4%B8%8E-FRiP-score-SPOT-score-%E7%9A%84%E8%AE%A1%E7%AE%97"><span class="nav-text">5. peak calling 与 FRiP_score&#x2F;SPOT_score 的计算</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E4%BA%8C%E3%80%81%E9%AB%98%E7%BA%A7%E5%88%86%E6%9E%90"><span class="nav-text">二、高级分析</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#1-IGV-%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="nav-text">1. IGV 可视化</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-TSS-TES-%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="nav-text">2. TSS&#x2F;TES 可视化</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-%E7%9B%B8%E5%85%B3%E6%80%A7%E5%8F%AF%E8%A7%86%E5%8C%96%EF%BC%88%E4%BE%9D%E6%97%A7%E6%98%AF%E4%BD%BF%E7%94%A8-deeptools%EF%BC%89"><span class="nav-text">3. 相关性可视化（依旧是使用 deeptools）</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#5-ChIPseeker-%E5%AF%B9-peaks-%E8%BF%9B%E8%A1%8C%E6%B3%A8%E9%87%8A%E5%92%8C%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="nav-text">5. ChIPseeker 对 peaks 进行注释和可视化</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#5-1-%E7%BB%93%E6%9E%84%E6%B3%A8%E9%87%8A"><span class="nav-text">5.1 结构注释</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#5-2-%E5%8A%9F%E8%83%BD%E6%B3%A8%E9%87%8A"><span class="nav-text">5.2 功能注释</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-homer-%E5%AF%BB%E6%89%BE-motif"><span class="nav-text">6. homer 寻找 motif</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#7-DiffBind-%E6%89%BE%E5%B7%AE%E5%BC%82-peak"><span class="nav-text">7. DiffBind 找差异 peak</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#8-footprint-%E5%88%86%E6%9E%90"><span class="nav-text">8. footprint 分析</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E4%B8%89%E3%80%81ATAC-seq-%E4%B8%8E%E5%A4%9A%E7%BB%84%E5%AD%A6%E6%95%B0%E6%8D%AE%E8%81%94%E5%90%88%E5%88%86%E6%9E%90"><span class="nav-text">三、ATAC-seq 与多组学数据联合分析</span></a></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Huang"
      src="/images/avatar4.png">
  <!--
  <p class="site-author-name" itemprop="name">Huang</p>
  -->
  <div class="site-description" itemprop="description">肿瘤/免疫/单细胞</div>
</div>
<div class="site-state-wrap animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
        <a href="/archives/">
          <span class="site-state-item-count">18</span>
          <span class="site-state-item-name">日志</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
          <a href="/categories/">
        <span class="site-state-item-count">15</span>
        <span class="site-state-item-name">分类</span></a>
      </div>
      <div class="site-state-item site-state-tags">
          <a href="/tags/">
        <span class="site-state-item-count">15</span>
        <span class="site-state-item-name">标签</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author animated">
      <span class="links-of-author-item">
        <a href="https://github.com/WhaleGe" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;WhaleGe" rel="noopener me" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:wdhuang927@gmail.com" title="Mail → mailto:wdhuang927@gmail.com" rel="noopener me" target="_blank"><i class="fa fa-envelope fa-fw"></i>Mail</a>
      </span>
  </div>


  <script type="text/javascript" charset="utf-8" src="/js/tagcloud.js"></script>
  <script type="text/javascript" charset="utf-8" src="/js/tagcanvas.js"></script>
  <div class="widget-wrap">
    <div id="myCanvasContainer" class="widget tagcloud">
        <canvas width="250" height="250" id="resCanvas" style="width=100%">
          <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/ATAC/" rel="tag">ATAC</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a><span class="tag-list-count">3</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Tfh/" rel="tag">Tfh</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/algorithm/" rel="tag">algorithm</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/bioinformatics/" rel="tag">bioinformatics</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/celltype/" rel="tag">celltype</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/essay/" rel="tag">essay</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/immune/" rel="tag">immune</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/knowledge/" rel="tag">knowledge</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a><span class="tag-list-count">3</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/note/" rel="tag">note</a><span class="tag-list-count">4</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/paper/" rel="tag">paper</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/pipeline/" rel="tag">pipeline</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/statistics/" rel="tag">statistics</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/tool/" rel="tag">tool</a><span class="tag-list-count">6</span></li></ul>
        </canvas>
    </div>
  </div>

        </div>
      </div>
        <div class="back-to-top animated" role="button" aria-label="返回顶部">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>
    </div>

    
  </aside>


    </div>

    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="https://yetu.fun/post/786e1faa.html">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar4.png">
      <meta itemprop="name" content="Huang">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Yetu">
      <meta itemprop="description" content="肿瘤/免疫/单细胞">
    </span>

    <span hidden itemprop="post" itemscope itemtype="http://schema.org/CreativeWork">
      <meta itemprop="name" content="ATAC分析流程 | Yetu">
      <meta itemprop="description" content="">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          ATAC分析流程
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">发表于</span>

      <time title="创建时间：2023-05-18 17:39:47" itemprop="dateCreated datePublished" datetime="2023-05-18T17:39:47+00:00">2023-05-18</time>
    </span>
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-folder"></i>
      </span>
      <span class="post-meta-item-text">分类于</span>
        <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
          <a href="/categories/notes/" itemprop="url" rel="index"><span itemprop="name">notes</span></a>
        </span>
          ，
        <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
          <a href="/categories/notes/ATAC/" itemprop="url" rel="index"><span itemprop="name">ATAC</span></a>
        </span>
    </span>

  
    <span class="post-meta-item" title="阅读次数" id="busuanzi_container_page_pv">
      <span class="post-meta-item-icon">
        <i class="far fa-eye"></i>
      </span>
      <span class="post-meta-item-text">阅读次数：</span>
      <span id="busuanzi_value_page_pv"></span>
    </span>
    <span class="post-meta-break"></span>
    <span class="post-meta-item" title="本文字数">
      <span class="post-meta-item-icon">
        <i class="far fa-file-word"></i>
      </span>
      <span class="post-meta-item-text">本文字数：</span>
      <span>7.8k</span>
    </span>
    <span class="post-meta-item" title="阅读时长">
      <span class="post-meta-item-icon">
        <i class="far fa-clock"></i>
      </span>
      <span class="post-meta-item-text">阅读时长 &asymp;</span>
      <span>28 分钟</span>
    </span>
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody"><blockquote>
<p>转载于网络，原文地址 <a target="_blank" rel="noopener" href="https://www.plob.org/article/24719.html">www.plob.org</a></p>
</blockquote>
<h3 id="一、基础知识"><a href="#一、基础知识" class="headerlink" title="一、基础知识"></a>一、基础知识</h3><h4 id="1-解密表观遗传学的三个方向与测序方法"><a href="#1-解密表观遗传学的三个方向与测序方法" class="headerlink" title="1. 解密表观遗传学的三个方向与测序方法"></a>1. 解密表观遗传学的三个方向与测序方法</h4><ul>
<li><strong>1. 探索染色质的开放性</strong>（chromatin accessibility）<br><code>ATAC-seq</code>: Assay of Transposase Accessible Chromatin sequencing<br><code>DNase-seq</code>: DNase I hypersensitive sites sequencing<br><code>FAIRE-seq</code>: Formaldehyde-Assisted Isolation of Regulatory Elements sequencing</li>
<li><strong>2. 探索转录因子的绑定与组蛋白修饰</strong>（TF binding and histone modifications）<br><code>ChIP-seq</code>: Chromatin Immuno-Precipitation sequencing</li>
<li><strong>3. 探索核小体占位</strong>（nucleosome positioning and occupancy）<br><code>MNase-seq</code>: Micrococcal Nuclease sequencing</li>
</ul>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632949992-700x506.png" alt=""></p>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632949994-700x844.png" alt=""></p>
<h4 id="2-名词解释"><a href="#2-名词解释" class="headerlink" title="2. 名词解释"></a>2. 名词解释</h4><ul>
<li><code>epigenetics</code>：表观遗传学。表观遗传学修饰在不改变 DNA 序列的情况下控制着基因的表达，包括染色质重塑、组蛋白修饰、DNA 甲基化和 microRNA 通路等</li>
<li><code>peaks</code>: 峰。常用来表示染色质的开放程度，因为是测序的 reads 落在了染色质的开放区，堆叠后被可视化的一种丰度的体现。</li>
</ul>
<p>但 peaks 到底代表什么？莫非 peaks 所在区域要么是启动子要么是增强子？再或者就是笼统定义为调控元件？（我目前的观点是，只能把他理解为酶切的位点，如果非要定义是否是特定分调控元件，只能进行后续分析。）</p>
<ul>
<li><p><code>THSs</code>: 转座酶超敏感位点（transposase hypersensitive sites）。</p>
</li>
<li><p><code>CREs</code>: 顺式调控元件（cis-regulatory elements）。即 DNA 分子中具有转录调节功能的特异 DNA 序列。按功能特性，真核基因顺式作用元件分为启动子、增强子及沉默子。</p>
</li>
<li><p><code>ACRs</code>: 染色质开放区域（accessible chromatin regions）。即正常或核小体被酶切裸露出来的 DNA 片段所在的区域。这篇<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41477-019-0548-z">【NP | 2019】</a>根据 ACRs 距离最近基因的距离将 ACRs 分为三种类型：genic (gACRs; overlapping a gene), proximal (pACRs; within 2kb of a gene) or distal (dACRs; &gt;2 kb from a gene)，分别是跨越基因的，近端的，远端的染色质开放区。</p>
</li>
<li><p><code>transposon</code>: 转座子。一段可以从原位上单独复制或断裂下来，环化后插入另一位点，并对其后的基因起调控作用的 DNA 序列。</p>
</li>
<li><p><code>promoter</code>: 启动子。启动子是位于结构基因 5’端上游的 DNA 序列，能活化 RNA 聚合酶，使之与模板 DNA 准确的结合并具有转录起始的特异性。每个启动子包括至少一个转录起始点以及一个以上的功能组件（典型的如 TATA 盒子）</p>
</li>
<li><p><code>proximal promoters</code>: 近端启动子。是 DNA 上位于基因开始之前的一个区域，在那里蛋白质和其他分子结合在一起准备读取该基因。</p>
</li>
<li><p><code>enhancer</code>: 增强子。增强子是远离转录起始点、决定基因的时间、空间特异性表达、增强启动子转录活性的 DNA 序列，其发挥作用的方式通常与方向、距离<strong>无关</strong>，可位于转录起始点的上游或下游。从功能上讲，没有增强子存在，启动子通常不能表现活性；没有启动子时，增强子也无法发挥作用。根据南京大学陈迪俊老师的研究表明增强子比启动子能结合更多的转录因子 <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-019-09513-2">(Nature Communications)</a></p>
</li>
<li><p><code>TFs</code>: 转录因子（transcription factors）是保证目的基因以特定的强度在特定的时间与空间表达的蛋白质分子。与 RNA 聚合酶 Ⅱ 形成转录起始复合体，共同参与转录起始的过程。</p>
</li>
<li><p><code>TSS</code>: 转录起始位点（transcription start site）。在一个典型的基因内部，排列顺序为转录起始位点 (TSS，一个碱基)- 起始密码子编码序列 (ATG)- 终止密码子编码序列 (TGA)- 转录终止位点 (TTS)，即 TSS-ATG-TGA-TTS</p>
</li>
<li><p><code>histone</code>：组蛋白。通常含有 H1，H2A，H2B，H3，H4 等 5 种成分，其中 H1 与 H3 极度富含赖氨酸（lysine），H1 不保守，其他组蛋白的基因非常保守。除 H1 外，其他 4 种组蛋白均分别以二聚体 (共八聚体）相结合，形成核小体核心。DNA 便缠绕在核小体的核心上。而 H1 则与核小体间的 DNA 结合</p>
</li>
<li><p><code>nucleosome</code>: 核小体。是由 DNA 和组蛋白形成的染色质基本结构单位。每个核小体由 146bp 的 DNA 缠绕组蛋白八聚体 1.75 圈形成。核小体核心颗粒之间通过 50bp 左右的连接 DNA 相连，暴露在核小体表面的 DNA 能被特定的核酸酶接近并切割</p>
</li>
<li><p><code>H3K4me1</code>: 组蛋白 H3 上的第 4 位赖氨酸发生单甲基化（mono-methylation of H3 at lysine 4） <a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41588-017-0018-3">（NG | H3K4me1 与增强子的关系）</a></p>
</li>
<li><p><code>H3K9ac</code>：组蛋白 H3 上的第 9 位赖氨酸发生乙酰化（H3 acetylation marks at lysine 9）关于 H3K4me1 与 H3K9ac 的不同，依旧可以参考这篇文献<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41477-019-0548-z">【NP | 2019】</a>或<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-019-09513-2">【NC | 2019】</a></p>
</li>
<li><p><code>组蛋白甲基化</code>：甲基化可发生在组蛋白的赖氨酸和精氨酸残基上，而且赖氨酸残基能够发生单、双、三甲基化，而精氨酸残基能够单、双甲基化，这些不同程度的甲基化极大地增加了组蛋白修饰和调节基因表达的复杂性</p>
</li>
<li><p><code>组蛋白乙酰化</code>：四种类型的组蛋白相互作用，将细胞核里的 DNA 紧紧地包装起来。这样的紧密包装能够有效阻止酶读取 DNA 上的遗传信息。然而，乙酰基连到组蛋白上能削弱它们对 DNA 的占据。因此局部乙酰化能暴露出相应的基因，让它们更容易激活<img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632949997-700x318.png" alt=""></p>
<p>  真核生物细胞核内染色质结构</p>
</li>
</ul>
<h3 id="二、基础分析"><a href="#二、基础分析" class="headerlink" title="二、基础分析"></a>二、基础分析</h3><h3 id="1-分析前的准备"><a href="#1-分析前的准备" class="headerlink" title="1. 分析前的准备"></a>1. 分析前的准备</h3><p>1.1 下载 ATAC 测序数据（或者用你自己的数据也行）</p>
<ul>
<li>所需软件：<a target="_blank" rel="noopener" href="https://www.ibm.com/aspera/connect/">Aspera Connect</a></li>
<li>软件安装与使用：参考<a target="_blank" rel="noopener" href="https://www.plob.org/tag/aspera">这系列文章</a></li>
<li>数据下载：项目 ID<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA480717/">【PRJNA480717】</a><a target="_blank" rel="noopener" href="https://academic.oup.com/jxb/article/70/15/3867/5481770">【文章原文】</a>，SRR 序列号如下图，下载方法和上文相同。由于数据量大，测试浪费时间，所以我在每个文件中取了 100w 条 reads，由于百度网盘太慢了，你可以去扣扣群 559758504 里面随便下载一个用于测试。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fastqc -t 8 -o ./ *.fastq.gz</span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950000-700x238.png" alt=""></p>
<p>image.png</p>
<p>1.2 下载 tair10 基因组与 gff3 注释文件</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bwa index -p /路径/bwaIndexTair /路径/TAIR10_genome.fa</span><br></pre></td></tr></table></figure>
<p>1.3 安装 miniconda</p>
<ul>
<li>由于后续分析大多数软件都是来源于 conda，因此你必须知道怎么使用。参考<a target="_blank" rel="noopener" href="https://www.plob.org/article/13100.html">【conda 的安装与使用】</a></li>
</ul>
<p>1.4 所需软件安装（分析中所用到的软件都汇集于此，下文不再介绍如何下载）</p>
<p>请先激活 conda 环境在安装！</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">conda install -c bioconda -y fastqc</span><br><span class="line">conda install -c bioconda -y trimmomatic</span><br><span class="line">conda install -c bioconda -y bwa</span><br><span class="line">conda install -c bioconda -y samtools</span><br><span class="line">conda install -c bioconda -y picard</span><br><span class="line">conda install -c bioconda -y macs2</span><br><span class="line">conda install -c bioconda -y bedtools</span><br><span class="line">conda install -c bioconda -y deeptools</span><br><span class="line">conda install -c bioconda -y homer</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>后续分析是使用 SRR7512044 作为测试数据，但是思路和方法都可以借鉴。</p>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950000-1.png" alt=""></p>
<p>测试数据</p>
<h3 id="2-测序质量检测"><a href="#2-测序质量检测" class="headerlink" title="2. 测序质量检测"></a>2. 测序质量检测</h3><ul>
<li>所需软件：<a target="_blank" rel="noopener" href="https://anaconda.org/bioconda/fastqc">FastQC</a><a target="_blank" rel="noopener" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">【官方网站】</a></li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fastqc -t 8 -o ./ *.fastq.gz</span><br></pre></td></tr></table></figure>
<p><code>-t</code>: 线程<br><code>-o</code>: 存放路径，不用指定前缀，默认为. fastq.gz 前面的字段<br><code>*.fastq.gz</code>: fastqc 可以同时接受多个 fastq.gz 文件，因此采用正则表达式【*】表示全部</p>
<ul>
<li>结果解读<br>由于开始测序时不准确，因此前 15bp 左右的 ATGC 含量有波动，需要在后面截掉。而最长 reads 为 76bp，则暗示着我最后至少要保留多长的序列。<br>一般我的计算思路是：保留的序列长度 =（最长的的 reads 长度 - 前面波动的碱基长度）× 40%</li>
</ul>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950001-700x514.png" alt=""></p>
<h3 id="2-测序质量检控制"><a href="#2-测序质量检控制" class="headerlink" title="2. 测序质量检控制"></a>2. 测序质量检控制</h3><ul>
<li>所需软件：<a target="_blank" rel="noopener" href="https://anaconda.org/bioconda/trimmomatic">Trimmomatic</a><a target="_blank" rel="noopener" href="https://links.jianshu.com/go?to=%255Bhttp%3A//www.usadellab.org/cms/%3Fpage%3Dtrimmomatic%255D%28http%3A//www.usadellab.org/cms/%3Fpage%3Dtrimmomatic%29">【官方网站】</a><a target="_blank" rel="noopener" href="https://www.plob.org/article/6258.html">【中文介绍】</a><br>其实这批数据已经做过质控了，如何判断的呢？QC 的为一个范围且短于提供给 NCBI 上的数据。因此我们没必要质控，但为了流程的完整性，还是假装质控一下。质控之后，还要进行质检，如此循环，直到合格。</li>
</ul>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950001-1-700x296.png" alt=""></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"># 定义一些变量，后面修改起来方便</span><br><span class="line">filename=SRR7512044</span><br><span class="line"> </span><br><span class="line">trimmomatic PE -threads 10 </span><br><span class="line">./$&#123;filename&#125;_1.fastq.gz ./$&#123;filename&#125;_2.fastq.gz </span><br><span class="line">./$&#123;filename&#125;_1.clean.fq.gz ./$&#123;filename&#125;_1.drop.fq.gz </span><br><span class="line">./$&#123;filename&#125;_2.clean.fq.gz ./$&#123;filename&#125;_2.drop.fq.gz </span><br><span class="line">HEADCROP:15 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:24</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><code>PE</code>: 双端模式。需要两个输入文件，正向测序序列和反向测序序列：<code>sample_R1.fastq.gz</code>和<code>sample_R2.fastq.gz</code>；<br>以及四个输出文件：<code>sample_1.clean.fq.gz</code>和<code>sample_1.drop.fq.gz</code>；<code>sample_2.clean.fq.gz</code>和<code>sample_2.drop.fq.gz</code><br><code>-threads</code>: 线程数<br><code>HEADCROP</code>: 从 reads 的开头切掉指定数量的碱基，即从 fastqc 中看的<br><code>LEADING</code>: 从 reads 的开头切除质量值低于阈值的碱基<br><code>TRAILING</code>: 从 reads 的末尾开始切除质量值低于阈值的碱基<br><code>SLIDINGWINDOW</code>: 从 reads 的 5’ 端开始，进行滑窗质量过滤，切掉碱基质量平均值低于阈值的滑窗<br><code>MINLEN</code>: 如果经过剪切后 reads 的长度低于阈值则丢弃这条 reads，即上面的计算方法：（76-15）×40%≈24</p>
<h3 id="3-建立索引与序列比对"><a href="#3-建立索引与序列比对" class="headerlink" title="3. 建立索引与序列比对"></a>3. 建立索引与序列比对</h3><ul>
<li>所需软件：<a target="_blank" rel="noopener" href="https://www.plob.org/tag/bwa">bwa</a> 和 <a target="_blank" rel="noopener" href="https://www.plob.org/tag/samtools">samtools</a></li>
<li>建立基因组索引。需要的时间挺长的，建议放后台。建好之后以后就可以跳过这一步了，且在经典的比对软件 STAR、hisat2、bowtie2 和 bwa-mem 中我更喜欢使用 STAR 和 bwa-mem。本次使用 bwa-mem 操作演示</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bwa index -p /路径/bwaIndexTair /路径/TAIR10_genome.fa</span><br></pre></td></tr></table></figure>
<p><code>-p</code>: 索引前缀，后缀自动补充<br><code>TAIR10_genome.fa</code>: 基因组文件</p>
<ul>
<li>序列比对<br>由于 sam 文件较大，因此我们直接跳过 sam，使用 samtools 转换为排序后的 bam 文件，注意【|】不要漏掉，且排序是依据 reads 比对到基因组的位置排的</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"># 定义一些变量，后面用</span><br><span class="line">filename=SRR7512044</span><br><span class="line">index=/路径/bwaIndexTair</span><br><span class="line"> </span><br><span class="line">bwa mem -M -t 8 </span><br><span class="line">-R &quot;@RGtID:$&#123;filename&#125;tSM:$&#123;filename&#125;tLB:WXStPL:Illumina&quot; </span><br><span class="line">$index </span><br><span class="line">$&#123;filename&#125;_1.clean.fq.gz </span><br><span class="line">$&#123;filename&#125;_2.clean.fq.gz </span><br><span class="line">| samtools sort -O bam -@ 10 -o ./$&#123;filename&#125;.raw.bam</span><br><span class="line">#无论干啥，生成bam文件后就要建立bam索引，并检查比对情况</span><br><span class="line">samtools index -@ 10 -b ./$&#123;filename&#125;.raw.bam</span><br><span class="line">samtools flagstat ./$&#123;filename&#125;.raw.bam &gt; ./$&#123;filename&#125;.raw.stat</span><br><span class="line">cat $&#123;filename&#125;.raw.stat</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><code>-M</code>: 将 shorter split hits 标记为次优，以兼容 Picard’s markDuplicates 软件<br><code>-R</code>: STR 完整的 read group 的头部，可以用’t’ 作为分隔符， 在输出的 SAM 文件中被解释为制表符 TAB. read group 的 ID，会被添加到输出文件的每一个 read 的头部，<br><code>-O</code>: 表示输出的 bam 文件<br><code>-o</code>: 输出的 bam 文件名<br><code>-t</code>与<code>@</code>: 线程数</p>
<ul>
<li>关于 stat 中内容的解释如下</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">14608455   0 in total (QC-passed reads   QC-failed reads) reads总数</span><br><span class="line">37967   0 secondary                                       出现比对到参考基因组多个位置的reads数</span><br><span class="line">0   0 supplementary                                       可能存在嵌合的reads数</span><br><span class="line">0   0 duplicates                                          重复的reads数</span><br><span class="line">14590894   0 mapped (99.88% : N/A)                        比对到参考基因组上的reads数</span><br><span class="line">14570488   0 paired in sequencing                         属于PE read的reads总数。</span><br><span class="line">7285244   0 read1                                         PE read中Read 1 的reads 总数。</span><br><span class="line">7285244   0 read2                                         PE read中Read 2 的reads 总数。</span><br><span class="line">14507068   0 properly paired (99.56% : N/A)               完美比对的reads总数。PE两端reads比对到同一条序列，且根据比对结果推断的插入片段大小符合设置的阈值。</span><br><span class="line">14551500   0 with itself and mate mapped                  PE两端reads都比对上参考序列的reads总数。</span><br><span class="line">1427   0 singletons (0.01% : N/A)                         PE两端reads，其中一端比上，另一端没比上的reads总数。</span><br><span class="line">26260   0 with mate mapped to a different chr             PE read中，两端分别比对到两条不同的序列的reads总数。</span><br><span class="line">17346   0 with mate mapped to a different chr (mapQ&gt;=5)   PE read中，两端分别比对到两条不同</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<h3 id="4-去除-PCR-重复并再次进行质控"><a href="#4-去除-PCR-重复并再次进行质控" class="headerlink" title="4. 去除 PCR 重复并再次进行质控"></a>4. 去除 PCR 重复并再次进行质控</h3><ul>
<li>所需软件：picard 与 <a target="_blank" rel="noopener" href="https://www.plob.org/tag/samtools">samtools</a><br>不建议使用 sambamba，虽然快，但是毛病多，尤其是在你不知情的情况下，软件卡死或者在最后输出文件时提示输出文件夹过多等。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">picard MarkDuplicates </span><br><span class="line">REMOVE_DUPLICATES=true </span><br><span class="line">I=$&#123;filename&#125;.raw.bam </span><br><span class="line">O=$&#123;filename&#125;.rmdup.bam </span><br><span class="line">M=$&#123;filename&#125;.rmdup.log</span><br><span class="line"># 实时监测我们的数据发生了什么变化</span><br><span class="line">samtools index $&#123;filename&#125;.rmdup.bam</span><br><span class="line">samtools flagstat $&#123;filename&#125;.rmdup.bam &gt; ./$&#123;filename&#125;.rmdup.stat</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><code>REMOVE_DUPLICATES=true</code>: 表示将检测出的 PCR duplication 直接去除；<br><code>I</code>: 或者 INPUT 指定输入的 bam 文件；<br><code>O</code>: 或者 OUTPUT 指定输出去除 PCR duplication 的 bam 文件；<br><code>M</code>: 或者 METRICS_FILE 指定输出的 matrix log 文件。</p>
<ul>
<li>下面是再次进行质控</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">samtools view -h -f 2 -q 30 $&#123;filename&#125;.rmdup.bam </span><br><span class="line">| grep -v -e &quot;mitochondria&quot; -e &quot;*&quot; -e &quot;chloroplast&quot; </span><br><span class="line">| samtools sort -O bam -@ 10 -o - &gt; $&#123;filename&#125;.last.bam</span><br><span class="line"># 实时监测我们的数据发生了什么变化</span><br><span class="line">samtools index $&#123;filename&#125;.last.bam</span><br><span class="line">samtools flagstat $&#123;filename&#125;.last.bam &gt; ./$&#123;filename&#125;.last.stat</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><code>-h</code>: 输出的 sam 文件带 header，默认不带<br><code>-f</code>: 输出含有所有 flag 的 reads<br><code>-q</code> 比对的最低质量值，一般 20/30 都可以<br>其他参数介绍看 <a target="_blank" rel="noopener" href="https://www.plob.org/tag/samtools">samtools 使用简述</a>，需要说明的是由于比对到线粒体和叶绿体的 reads 不是我们关注的重点，因此使用<code>grep</code>剔除掉，当然了，做动物的话需要剔除什么视情况而定了。</p>
<ul>
<li>好了，下面来看看我们的数据从 raw data -&gt; rmdup -&gt; last 到底发生了那些变化<img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950002-700x344.png" alt=""></li>
<li>然后，从 IGV 看看发生了什么变化<br>关于使用 IGV 查看 bam 文件中 reads 呈现不同颜色的说明</li>
</ul>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950005-700x773.png" alt=""></p>
<h3 id="5-统计过滤后的结果中-Insertsize-长度分布"><a href="#5-统计过滤后的结果中-Insertsize-长度分布" class="headerlink" title="5. 统计过滤后的结果中 Insertsize 长度分布"></a>5. 统计过滤后的结果中 Insertsize 长度分布</h3><ul>
<li>所需软件：picard<br>Insertsize =（read2 的比对位点 - read1 的比对位点） read2 的长度。也可以随便点击上图 IGV 中任一一条 reads 查看插入片段的大小</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">picard CollectInsertSizeMetrics </span><br><span class="line">I=./$&#123;filename&#125;.last.bam </span><br><span class="line">O=./$&#123;filename&#125;.last.insertsize </span><br><span class="line">H=./$&#123;filename&#125;.last.hist.pdf</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950006-700x683.png" alt=""></p>
<h3 id="5-peak-calling-与-FRiP-score-SPOT-score-的计算"><a href="#5-peak-calling-与-FRiP-score-SPOT-score-的计算" class="headerlink" title="5. peak calling 与 FRiP_score/SPOT_score 的计算"></a>5. peak calling 与 FRiP_score/SPOT_score 的计算</h3><ul>
<li>所需软件：<a target="_blank" rel="noopener" href="https://www.plob.org/article/7118.html">macs2</a> 和 <a target="_blank" rel="noopener" href="https://www.plob.org/tag/bedtools">bedtools</a><br>因为 macs2 需要用到基因组的有效大小，而软件本身就提供了 4 个物种的有效基因组大小。</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">effective_genome_size=119481543</span><br><span class="line"> </span><br><span class="line">bedtools bamtobed -i $&#123;filename&#125;.last.bam &gt; $&#123;filename&#125;.last.bed</span><br><span class="line">macs2 callpeak </span><br><span class="line">-t $&#123;filename&#125;.last.bed </span><br><span class="line">-g $&#123;effective_genome_size&#125; </span><br><span class="line">--nomodel --shift -100 --extsize 200 </span><br><span class="line">-n $&#123;filename&#125;.last </span><br><span class="line">-q 0.01 --outdir ./peaks</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<ul>
<li>其实，这里面讨论最多的是<code>--nomodel --shift -100 --extsize 200</code>这些参数如何选择，下面的图很形象的展示了参数的作用。当然，我也是查阅了很多资料与文献，一般默认在<code>ATAC-seq，DNase-seq，FAIRE-seq</code>的时候将<code>shift</code>设置为<code>extsize</code>的一半，且参数固定为：<code>--nomodel --shift 100 --extsize 200</code>。而在<code>MNase-seq</code>的时候，参数固定为：<code>--nomodel --shift 37 --extsize 73</code>。在<code>ChiP-seq</code>的时候不用移峰，所以只使用<code>-nomodel</code>，当做组蛋白修饰的时候，由于 peak 并<a target="_blank" rel="noopener" href="http://blog.sina.com.cn/s/blog_4bc742520102uy64.html">不典型</a>，所以使用<code>–nomodel –shift 73</code>参数。另外如何使用<code>macs2 predictd</code>预测双峰模型，这里就不展开介绍了，可以私聊我，我把脚本发你。</li>
</ul>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950007.png" alt=""></p>
<p>shift 与 extsize 到底在干啥？</p>
<ul>
<li>下面是我见过的图中解释<strong>移峰</strong>现象最清楚的了，可以参考</li>
</ul>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950007-1-700x503.png" alt=""></p>
<p>移峰</p>
<ul>
<li>最后，就是输出文件的解读<ol>
<li><strong>${filename}_peaks.xls</strong></li>
</ol>
</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">前几行是callpeak时的命令。后面是：</span><br><span class="line">1.染色体号</span><br><span class="line">2.peak起始位点</span><br><span class="line">3.peak结束位点</span><br><span class="line">4.peak区域长度</span><br><span class="line">5.peak的峰值位点（summit position）</span><br><span class="line">6.peak 峰值的高度（pileup height at peak summit, -log10(pvalue) for the peak summit）</span><br><span class="line">7.peak的富集倍数（相对于random Poisson distribution with local lambda）</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<ul>
<li><ol>
<li><strong>${filename}_peaks.narrowPeak</strong></li>
</ol>
</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">narrowPeak文件是BED6 4格式，可以上传到IGV查看</span><br><span class="line">1.染色体号</span><br><span class="line">2.peak起始位点</span><br><span class="line">3.peak结束位点</span><br><span class="line">4.peak name（样本名 _peak_1的后缀之类的）</span><br><span class="line">5.-10*log10qvalue</span><br><span class="line">6.正负链</span><br><span class="line">7.fold change</span><br><span class="line">8.-log10pvalue</span><br><span class="line">9.-log10qvalue</span><br><span class="line">10.相对于峰的峰顶位置（relative summit position to peak start）</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<ul>
<li>注意：excel 里的起始坐标需要减 1 才与 narrowPeak 的坐标一致，见下图。<ol>
<li><strong>${filename}_summits.bed</strong></li>
</ol>
</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">BED格式的文件，包含peak的summits（峰顶）位置</span><br><span class="line">1.染色体号</span><br><span class="line">2.峰的峰顶起始位点（注意是峰顶，不是峰，他只是一个点）</span><br><span class="line">3.峰的峰顶起始位点（因此二者就差一个碱基的位置）</span><br><span class="line">4.峰顶（样本名 _peak_1的后缀之类的）</span><br><span class="line">5.-log10pvalue</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<ul>
<li><p>注意：excel 里的起始坐标需要减 1 才与 narrowPeak 的坐标一致，见下图。</p>
</li>
<li><p><strong>另外，需要告诉你的是，bed 文件中第 2 列中的峰顶起始位点是如何计算的？</strong><br>  bed 文件中第 2 列（峰的峰顶起始位点） = xls 第二列 xls 第十列 -1。或者↓↓↓↓<br>  bed 文件中第 2 列（峰的峰顶起始位点） = narrowPeak 第二列 narrowPeak 第十列</p>
</li>
<li><p>总览比较：<img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950008-700x407.png" alt=""></p>
<p>  bed，narrowPeak，bdg，xls 四种类型输出文件的比较</p>
</li>
<li><p>FRiP_score 的计算<a target="_blank" rel="noopener" href="https://www.encodeproject.org/atac-seq/">【官方介绍】</a><br>  简单来说，FRiP_score 说的就是统计落在 Peaks 里面的 reads 数量与所有比对到基因组上 reads 数量的比值。可以作为判定样本测序质量的一个指标。</p>
</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">Reads=$(bedtools intersect -a $&#123;filename&#125;.last.bed -b $&#123;filename&#125;_peaks.narrowPeak |wc -l|awk &#x27;&#123;print $1&#125;&#x27;)</span><br><span class="line">totalReads=$(wc -l $&#123;filename&#125;.last.bed|awk &#x27;&#123;print $1&#125;&#x27;) </span><br><span class="line">echo $&#123;Reads&#125; $&#123;totalReads&#125; $&#123;filename&#125; &#x27;FRiP_score:&#x27; $(echo &quot;scale=2;100*$&#123;Reads&#125;/$&#123;totalReads&#125;&quot;|bc)&#x27;%&#x27;</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<ul>
<li>官网上说 ATAC-Seq 的 FRiP score should be &gt;30%（<a target="_blank" rel="noopener" href="https://www.encodeproject.org/atac-seq/">https://www.encodeproject.org/atac-seq/</a>）我猜是如果大多数 reads 都不在 peaks 区域，说明测序质量或者实验效果不是太好，因为酶切切开的就是 peaks 区域，太少不是就偏离实验目的了嘛</li>
<li>SPOT_score 的计算<a target="_blank" rel="noopener" href="https://www.encodeproject.org/data-standards/terms/">【官方介绍】</a><a target="_blank" rel="noopener" href="https://www.encodeproject.org/software/hotspot/">【软件安装】</a><br>简单来说，SPOT_score 就是在 FRiP_score 的基础上增加了测序深度的条件限制，比如 FRiP 是统计全局的 reads，而 SPOT 只统计测序深度为 30× 的 reads，大于 30× 的 reads 不再参与统计。目前我还在探索中如何计算。</li>
</ul>
<h3 id="二、高级分析"><a href="#二、高级分析" class="headerlink" title="二、高级分析"></a>二、高级分析</h3><p>所谓的高级并不高大上，做了有可能也不能让你的文章提升档次，望周知！</p>
<h3 id="1-IGV-可视化"><a href="#1-IGV-可视化" class="headerlink" title="1. IGV 可视化"></a>1. IGV 可视化</h3><ul>
<li>前面的 bam 文件，narrowPeak 文件，bed 文件都有了，现在还差一个 bw 文件，因此我们先跑下面的代码，然后在 IGV 里面统一看一下</li>
<li>所需软件：<a target="_blank" rel="noopener" href="https://anaconda.org/bioconda/deeptools">deeptools</a> 和 <a target="_blank" rel="noopener" href="https://software.broadinstitute.org/software/igv/download">IGV</a><br>主要是为生成 bw 文件用于 IGV 展示</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">bamCoverage --bam $&#123;filename&#125;.last.bam -o $&#123;filename&#125;.last.bw </span><br><span class="line">    --binSize 10 </span><br><span class="line">    --normalizeUsing RPGC </span><br><span class="line">    --effectiveGenomeSize $&#123;effective_genome_size&#125; </span><br><span class="line">    --ignoreForNormalization chrX </span><br><span class="line">    --extendReads</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>注：如果发现没有这个包 pysam.libchtslib，直接安装<code>pip install numpydoc</code><br><code>-o</code>: 生成 bw 文件<br><code>binSize</code>: 自行设定覆盖度计算的窗口大小 (bin)<br><code>normalizeUsing</code>: 数据准化方法，有三种 RPGC，RPKM，CPM，BPM<br><code>ignoreForNormalization</code>: 设置你想忽略的染色体名称<br><code>extendReads</code>: 让 reads 扩展至 fragment 的大小</p>
<ul>
<li>IGV 一览</li>
</ul>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950009-700x436.png" alt=""></p>
<p>IGV 可视化 bw、narrowpeak，bed，bam，gff 文件</p>
<h3 id="2-TSS-TES-可视化"><a href="#2-TSS-TES-可视化" class="headerlink" title="2. TSS/TES 可视化"></a>2. TSS/TES 可视化</h3><ul>
<li><p>所需软件：<a target="_blank" rel="noopener" href="https://anaconda.org/bioconda/deeptools">deeptools</a><br>  主要是如何生成 <a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/2494.html">Refseq 文件</a>，虽然官网上提供了一些，但总归是不能覆盖所有物种，因此，我们可以从 gff/gtf 文件中自己创造。</p>
</li>
<li><p>先提取<strong>基因</strong>的位置信息（不是 cds 或者 exon），根据不同的 gff 进行修改，但格式始终是<code>`chrom/chromStart/chromEnd/name/score/strand` </code></p>
  <figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">awk -vFS=&#x27;[\t=;]&#x27; -vOFS=&quot;\t&quot; \</span><br><span class="line">&#x27;&#123;if ($3==&quot;gene&quot;) print $1,$4,$5,$10,$6,$7&#125;&#x27; \</span><br><span class="line">TAIR10.gff|sed &#x27;s/Chr//g&#x27; &gt; refseq.bed  </span><br><span class="line">#由于tair官网上提供的基因组文件中染色人命名没有Chr，为保持一致，这里去掉</span><br><span class="line"></span><br></pre></td></tr></table></figure>
</li>
<li><p>然后进行 TSS 可视化</p>
</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">computeMatrix reference-point --referencePoint TSS  -p 15 </span><br><span class="line">    -b 2000 -a 2000 </span><br><span class="line">    -R ./refseq.bed </span><br><span class="line">    -S *.bw </span><br><span class="line">    -o TSS.gz </span><br><span class="line">    --skipZeros </span><br><span class="line">    --outFileSortedRegions Heatmap1sortedRegions.bed</span><br><span class="line">plotHeatmap -m TSS.gz -out TSS.Heatmap.pdf --plotFileFormat pdf</span><br><span class="line">plotProfile -m TSS.gz -out TSS.Profile.pdf --plotFileFormat pdf --perGroup</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950010-700x629.png" alt=""></p>
<ul>
<li>然后进行 TES 可视化</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">computeMatrix scale-regions  -p 15  </span><br><span class="line">    -b 2000 -a 2000 </span><br><span class="line">    -R ./refseq.bed </span><br><span class="line">    -S *.bw </span><br><span class="line">    --skipZeros -o body.gz</span><br><span class="line">plotHeatmap -m body.gz -out body.Heatmap.pdf --plotFileFormat pdf</span><br><span class="line">plotProfile -m body.gz -out body.Profile.pdf --plotFileFormat pdf --perGroup</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950011-700x630.png" alt=""></p>
<ul>
<li>另外，参考下面这个图，看每个参数控制的哪里，如何同时绘制多个等。这是我好久之前总结的，清晰度堪忧</li>
</ul>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950013-700x649.png" alt=""></p>
<h3 id="3-相关性可视化（依旧是使用-deeptools）"><a href="#3-相关性可视化（依旧是使用-deeptools）" class="headerlink" title="3. 相关性可视化（依旧是使用 deeptools）"></a>3. 相关性可视化（依旧是使用 deeptools）</h3><ul>
<li>因为我测试样本少，所以就使用<a target="_blank" rel="noopener" href="https://deeptools.readthedocs.io/en/latest/content/tools/plotCorrelation.html">官网的图</a>了哈，但代码是能跑通的，放心使用。</li>
<li>先使用 [bw] 文件生成 [npz] 文件。所有样本做相关性，用 [*.last.bw]，多个单列出来也行</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">multiBigwigSummary bins -b *.last.bw -o number.of.bins.npz</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<ul>
<li>画相关性热图</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">plotCorrelation -in number.of.bins.npz </span><br><span class="line">--corMethod pearson </span><br><span class="line">--skipZeros </span><br><span class="line">--plotTitle &quot;Pearson Correlation of Average Scores Per Transcript&quot; </span><br><span class="line">--plotFileFormat pdf </span><br><span class="line">--whatToPlot heatmap --colorMap RdYlBu --plotNumbers </span><br><span class="line">-o heatmap_PearsonCorr_bigwigScores.pdf </span><br><span class="line">--outFileCorMatrix PearsonCorr_bigwigScores.tab</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950014-700x605.png" alt=""></p>
<p>heatmap</p>
<ul>
<li>画相关性点图</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">plotCorrelation -in number.of.bins.npz </span><br><span class="line">--corMethod pearson </span><br><span class="line">--skipZeros </span><br><span class="line">--plotTitle &quot;Pearson Correlation of Average Scores Per Transcript&quot; </span><br><span class="line">--plotFileFormat pdf </span><br><span class="line">--whatToPlot scatterplot </span><br><span class="line">-o scatterplot_PearsonCorr_bigwigScores.pdf </span><br><span class="line">--outFileCorMatrix PearsonCorr_bigwigScores.tab</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950015-700x734.png" alt=""></p>
<p>相关性点图</p>
<h3 id="5-ChIPseeker-对-peaks-进行注释和可视化"><a href="#5-ChIPseeker-对-peaks-进行注释和可视化" class="headerlink" title="5. ChIPseeker 对 peaks 进行注释和可视化"></a>5. ChIPseeker 对 peaks 进行注释和可视化</h3><ul>
<li>对 peak 的注释分为两个部分——结构注释和功能注释<br>结构注释会将 peak 所落在基因组上的区域结构注释出来，比如说启动子区域，UTR 区域，内含子区域等等。同时，也会将与 peak 最临近的基因注释出来，非常好用。<br>功能注释其实并不是对 peak 进行注释的，而是对 peak 最临近的基因注释的，因此这部分就和传统的 GO/KEGG 等分析如出一辙啦。</li>
</ul>
<h4 id="5-1-结构注释"><a href="#5-1-结构注释" class="headerlink" title="5.1 结构注释"></a><strong>5.1 结构注释</strong></h4><p>5.1.1 <strong>准备工作</strong>：导入包，没有的安装吧。另外需要使用 gff/gtf 构建一个 TxDb。因为<a target="_blank" rel="noopener" href="http://www.bioconductor.org/packages/release/BiocViews.html%23___TxDb">官网</a>上 TxDb 也不是每一个物种都有（目前共 43 个），因此我们自己手动构建。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">#if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))</span><br><span class="line">#    install.packages(&quot;BiocManager&quot;)</span><br><span class="line">#BiocManager::install(&quot;clusterProfiler&quot;)</span><br><span class="line"> </span><br><span class="line">library(clusterProfiler)</span><br><span class="line">library(ChIPseeker)</span><br><span class="line">library(GenomicFeatures)</span><br><span class="line">txdb &lt;- makeTxDbFromGFF(&quot;public/bioinfoYu/genome/tair10/TAIR10.gff&quot;,</span><br><span class="line">                        format=&quot;gff&quot;)    #也可以使用gtf</span><br><span class="line">keytypes(txdb)    #感兴趣的话，可以用以下方法探索txdb都包含了什么内容</span><br><span class="line">keys(txdb)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>5.1.2 <strong>单个文件的结构注释</strong>（不推荐在这一步加 OrgDB 的内容，没用）</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">#读入单个summits文件</span><br><span class="line">peaks &lt;- readPeakFile(&quot;SRR7512044_summits.bed&quot;)</span><br><span class="line">#结构注释</span><br><span class="line">peakAnno &lt;- annotatePeak(peaks,</span><br><span class="line">                         TxDb=txdb,</span><br><span class="line">                         tssRegion=c(-1000, 1000))</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>上面的 tssRegion 取值有说法的，因为判定 “Promoter”, “Exon”, “Intron”, “Intergenic” 等结构的依据就是根据范围来的。比如定义 1000 以内的，则在 1kbp 之内规定为 “Promoter (&lt;=1kb)”，而 1k-2k 内定义为 “Downstream (1-2kb)”。但你定义 2000 以内呢，则在 1kbp 之内规定为 “Promoter (&lt;=1kb)”，而 1k-2k 内定义为 “Promoter (1-2kb)”。虽然，ChIPseeker 会标记出来区域，但还是三思后再取范围。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">#注释完，进行可视化，多种图可供选择</span><br><span class="line">plotAnnoBar(peakAnno)</span><br><span class="line">plotDistToTSS(peakAnno)</span><br><span class="line">vennpie(peakAnno)</span><br><span class="line">plotAnnoPie(peakAnno)</span><br><span class="line">#install.packages(&quot;ggupset&quot;)</span><br><span class="line">library(ggupset)</span><br><span class="line">upsetplot(peakAnno)</span><br><span class="line">#install.packages(&quot;ggimage&quot;)</span><br><span class="line">library(ggimage)</span><br><span class="line">upsetplot(peakAnno, vennpie=TRUE)</span><br><span class="line"> </span><br><span class="line">#最后将我们的注释结果转为数据框，便于查看</span><br><span class="line">df &lt;- as.data.frame(peakAnno)</span><br><span class="line">#将注释到的基因提取出来（第14列），用于后续功能分析</span><br><span class="line">gene &lt;- df[,14]</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950016-700x323.png" alt=""></p>
<p>提供可视化的方法</p>
<p>5.1.3 <strong>多个文件的结构注释</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">#一次也可以读入多个summits文件，使用list存储，然后使用lapply注释</span><br><span class="line">files = list(peak1 = &quot;peaks.bed&quot;, peak2 = (&quot;peak.bed&quot;))</span><br><span class="line">peakAnnoList &lt;- lapply(files, </span><br><span class="line">                       annotatePeak,</span><br><span class="line">                       TxDb=txdb,</span><br><span class="line">                       tssRegion=c(-2000, 2000))</span><br><span class="line">plotAnnoBar(peakAnnoList)</span><br><span class="line">plotDistToTSS(peakAnnoList)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950019-700x220.png" alt=""></p>
<p>读入多个 summits 文件的可视化</p>
<h4 id="5-2-功能注释"><a href="#5-2-功能注释" class="headerlink" title="5.2 功能注释"></a><strong>5.2 功能注释</strong></h4><p>5.2.1 <strong>可选支线</strong><br>由于这里需要物种的 OrgDb 注释库。如果你足够幸运，研究的物种正好在<a target="_blank" rel="noopener" href="http://www.bioconductor.org/packages/release/BiocViews.html%23___OrgDb">官网</a>仅提供的 20 个物种中存在，那就拿来用吧。除了以上这些都好说，还有是 id 转换，后面再说。<br>5.2.2 <strong>OrgDb 下载</strong><br>本次测试是用拟南芥的样本，我推荐在<a target="_blank" rel="noopener" href="http://www.bioconductor.org/packages/release/BiocViews.html%23___OrgDb">官网</a>中直接将所需要的的 OrgDb 库下载到服务器上，这样就不用在 R 里面重复在网上获取浪费时间了。比如说拟南芥，就可以下载，这样就把【org.At.tair.db】下载到了服务器或者本地了。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">wget http://www.bioconductor.org/packages/release/data/annotation/src/contrib/org.At.tair.db_3.13.0.tar.gz</span><br><span class="line">tar -zvxf org.At.tair.db_3.13.0.tar.gz</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>5.2.3 <strong>导入 OrgDb</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">#从本地下载OrgDb库（如果你直接下载到了R的lib里面，就直接library也可）</span><br><span class="line">install.packages(&quot;/public/bioinfoYu/genome/tair10/org.At.tair.db&quot;, repos=NULL)</span><br><span class="line">#另外一种方法</span><br><span class="line">#if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))install.packages(&quot;BiocManager&quot;)</span><br><span class="line">#BiocManager::install(&quot;org.At.tair.db&quot;)</span><br><span class="line">#载入库</span><br><span class="line">library(org.At.tair.db)</span><br><span class="line">#感兴趣的话，看看org.At.tair.db包含的都是些啥</span><br><span class="line">#查看包含的列名</span><br><span class="line">keytypes(org.At.tair.db)</span><br><span class="line">#查看包含的基因ID</span><br><span class="line">keys(org.At.tair.db)</span><br><span class="line">#随便选择一个基因和一些列名，看一看到第是什么东东，不过因为导入了其他包，下面命令报错，我至今还没有解决</span><br><span class="line">select(org.At.tair.db,keys=&quot;AT1G01010&quot;,columns = c(&quot;ENTREZID&quot;,&quot;GO&quot;,&quot;SYMBOL&quot;))</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>5.2.4 <strong>ID 转换</strong>（非必须）<br>上面<code>5.1.2</code>已经得到 peak 附近的基因集【gene】了，如果需要 ID 转换，就是用<code>clusterProfiler</code>中的<code>bitr</code>函数</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">#首先，去除版本号，如AT1G50040.1后面的.1，当然我们的测试集没有，因此不用做</span><br><span class="line">#gene_rm &lt;- gsub(&quot;.[0-9] $&quot;,&quot;&quot;,gene)</span><br><span class="line">#下面进行ID转换</span><br><span class="line">keytypes(org.At.tair.db)</span><br><span class="line">gene_last &lt;- bitr(geneID = gene, </span><br><span class="line">                    fromType = &quot;TAIR&quot;,</span><br><span class="line">                    toType = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),</span><br><span class="line">                    OrgDb = org.At.tair.db)</span><br><span class="line"> </span><br><span class="line"># geneID    需要转换的ID</span><br><span class="line"># fromType  当前ID类型</span><br><span class="line"># toType    转换成什么ID，使用keytypes()查看有哪些类型</span><br><span class="line"># OrgDb     注释数据库</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>5.2.5 <strong>GO 功能注释</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"># 因为keyType不接受数据框类型的数据，因此我们转换为字符型</span><br><span class="line">genelist &lt;- gene.symbol[,2]  #注意我这里把ENTREZID单独拿出来了，所以下面自然是keyType = &quot;ENTREZID&quot;,</span><br><span class="line">go_result &lt;- enrichGO(genelist,</span><br><span class="line">                      keyType = &quot;ENTREZID&quot;, </span><br><span class="line">                      OrgDb = org.At.tair.db, </span><br><span class="line">                      ont = &quot;BP&quot;, #BP,MF,CC,ALL可选</span><br><span class="line">                      pAdjustMethod = &quot;BH&quot;, </span><br><span class="line">                      qvalueCutoff = 0.05, </span><br><span class="line">                      readable = FALSE)</span><br><span class="line">#可视化，showCategory=20调整显示的条目多少</span><br><span class="line">dotplot(go_result, showCategory=20)</span><br><span class="line">#将结果转为数据框，可以选择导出</span><br><span class="line">goresult &lt;- as.data.frame(go_result)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950021-700x685.png" alt=""></p>
<p>GO 可视化</p>
<p>5.2.6 <strong>KEGG 通路注释</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">#geneClusters必须是数据框类型，ID必须是ENTREZID类型</span><br><span class="line">class(gene.symbol)</span><br><span class="line">compKEGG &lt;- compareCluster(geneClusters=gene.symbol, </span><br><span class="line">                           fun = &quot;enrichKEGG&quot;,</span><br><span class="line">                           organism = &quot;ath&quot;,    #在 http://www.genome.jp/kegg/catalog/org_list.html 上查看物种简写</span><br><span class="line">                           pvalueCutoff  = 0.05, </span><br><span class="line">                           pAdjustMethod = &quot;BH&quot;)</span><br><span class="line">dotplot(compKEGG, showCategory = 20, title = &quot;KEGG Pathway Enrichment Analysis&quot;)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950022-700x342.png" alt=""></p>
<p>KEGG 可视化</p>
<h3 id="6-homer-寻找-motif"><a href="#6-homer-寻找-motif" class="headerlink" title="6. homer 寻找 motif"></a>6. homer 寻找 motif</h3><ul>
<li>所需软件：<a target="_blank" rel="noopener" href="https://anaconda.org/bioconda/homer">homer</a><a target="_blank" rel="noopener" href="http://homer.ucsd.edu/homer/">【官方网站】</a></li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"># 提取peaks的位置信息文件</span><br><span class="line">awk &#x27;&#123;print $4&quot;t&quot;$1&quot;t&quot;$2&quot;t&quot;$3&quot;t &quot;&#125;&#x27; </span><br><span class="line">SRR7512044_peaks.narrowPeak &gt; SRR7512044.homer_peaks.tmp</span><br><span class="line"># 寻找motif</span><br><span class="line">findMotifsGenome.pl SRR7512044.homer_peaks.tmp </span><br><span class="line">./TAIR10_genome.fa   #指定参考基因组</span><br><span class="line">./SRR7512044.motif   #输出文件夹的名字</span><br><span class="line">-size 200 -len 8,10,12</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950023-700x261.png" alt=""></p>
<h3 id="7-DiffBind-找差异-peak"><a href="#7-DiffBind-找差异-peak" class="headerlink" title="7. DiffBind 找差异 peak"></a>7. DiffBind 找差异 peak</h3><ul>
<li>所需软件：<a target="_blank" rel="noopener" href="http://www.bioconductor.org/packages/release/bioc/html/DiffBind.html">DiffBind</a></li>
</ul>
<p>首先根据我自己的理解将<strong>找差异 peak 的原理</strong>说一下：既然是找差异的 peak，那首先就要保证不同样本间 peak 所在的区间基本一致才可以。因此<strong>第一步</strong>就是找到多个样本相同的 peak 区间。<strong>第二步</strong>，基于这些相同区域的 peak 开始找差异，这个地方的差异和传统的 RNA-Seq 找差异基因的原理一模一样，都是<strong>对落在峰区间的 reads 进行计数</strong>，数量不同的话差异就由此而来。但软件都帮我们做了，就不用考虑那么多了</p>
<ul>
<li>我废话真多，其实最麻烦的就是<strong>准备输入数据</strong>，下面来看怎么做<br>一般情况相下需要包括以下几列：<code>&quot;SampleID&quot;， &quot;Tissue&quot;， &quot;Factor&quot;， &quot;Condition&quot; ，&quot;Treatment&quot;，&quot;Replicate&quot;， &quot;bamReads&quot; ，&quot;ControlID&quot; ，&quot;bamControl&quot; ，&quot;Peaks&quot;和&quot;PeakCaller&quot;</code></li>
</ul>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950025-700x148.png" alt=""></p>
<ul>
<li>但我们做 ATAC-Seq 的哪里有什么<code>control，controlID，bamControl</code>，因此删掉，而<code>&quot;Tissue&quot;， &quot;Factor&quot;， &quot;Condition&quot; ，&quot;Treatment&quot;，&quot;Replicate&quot;，</code>表示的都是分组的意思，因此我保留了两个<code>&quot;Factor&quot;，&quot;Replicate&quot;</code>，最后 csv 文件的样子如下：<img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950026-700x127.png" alt=""></li>
</ul>
<p><code>SampleID</code>：不能有重复<br><code>Factor</code>：进行分组，后面要做差异分析的依据<br><code>Replicate</code>：重复的编号<br><code>bamReads</code>：最后生成的 bam 文件。全路径 bam 文件名<br><code>Peaks</code>：macs2 生成的 narrowPeak 文件。全路径 narrowPeak 文件名<br><code>PeakCaller</code>：说明 peak 类型，使用 narrowPeak 文件则为<code>narrow</code>。使用 xls 文件，则用<code>macs</code>。使用 bed 文件，则用<code>bed</code></p>
<ul>
<li>开始正题</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"># 安装包</span><br><span class="line">if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))install.packages(&quot;BiocManager&quot;)</span><br><span class="line">BiocManager::install(&quot;DiffBind&quot;)</span><br><span class="line">library(DiffBind)</span><br><span class="line">#导入数据，dba函数会将会将bam文件一同导入，因此csv中的路径一定要准确</span><br><span class="line">tamoxifen &lt;- dba(sampleSheet=&quot;atDiff.csv&quot;)</span><br><span class="line">#查看样本间的相关性</span><br><span class="line">plot(tamoxifen)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950027.png" alt=""></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">#计算每个样本中的reads数</span><br><span class="line">tamoxifen &lt;- dba.count(tamoxifen)</span><br><span class="line">#简单查看计算结果</span><br><span class="line">info &lt;- dba.show(tamoxifen)</span><br><span class="line">libsizes &lt;- cbind(LibReads=info$Reads, FRiP=info$FRiP, PeakReads=round(info$Reads * info$FRiP))</span><br><span class="line">rownames(libsizes) &lt;- info$ID</span><br><span class="line">libsizes</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">#统计结果</span><br><span class="line">       LibReads FRiP PeakReads</span><br><span class="line">cmt2_1  6720462 0.29   1948934</span><br><span class="line">cmt2_2  7747491 0.33   2556672</span><br><span class="line">wild_1  7078309 0.36   2548191</span><br><span class="line">wild_2  8030752 0.34   2730456</span><br><span class="line">ddcc_1  6370232 0.30   1911069</span><br><span class="line">ddcc_2  6956429 0.35   2434750</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p>&lt;pre”&gt;# 默认基于测序深度对数据进行标椎化 tamoxifen &lt;- dba.normalize(tamoxifen) norm &lt;- dba.normalize(tamoxifen, bRetrieve=TRUE) norm #查看进行标准化的文库大小 (library-size)，其实就是各样本文库大小总和的平均值，你可以计算验证一下 normlibs &lt;- cbind(FullLibSize=norm$lib.sizes, NormFacs=norm$norm.factors, NormLibSize=round(norm$lib.sizes/norm$norm.factors)) rownames(normlibs) &lt;- info$ID normlibs</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">#标椎化结果</span><br><span class="line">       FullLibSize  NormFacs NormLibSize</span><br><span class="line">cmt2_1     6720462 0.9398443     7150612</span><br><span class="line">cmt2_2     7747491 1.0834724     7150612</span><br><span class="line">wild_1     7078309 0.9898885     7150612</span><br><span class="line">wild_2     8030752 1.1230859     7150612</span><br><span class="line">ddcc_1     6370232 0.8908652     7150612</span><br><span class="line">ddcc_2     6956429 0.9728438     7150612</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">#分组，格式是表头在最前面，要分的组依次写在后面，只能两两比较，因此后面只能写两组，但可以多执行几次，都会追加到tamoxifen 中</span><br><span class="line">#分组1，后面使用contrast=1单独查看</span><br><span class="line">tamoxifen &lt;- dba.contrast(tamoxifen, contrast=c(&quot;Factor&quot;,&quot;ddcc&quot;,&quot;wild&quot;))</span><br><span class="line">#分组2，后面使用contrast=2单独查看</span><br><span class="line">tamoxifen &lt;- dba.contrast(tamoxifen, contrast=c(&quot;Factor&quot;,&quot;cmt2&quot;,&quot;wild&quot;))</span><br><span class="line">#当然还可有分组3,4,5等，均可以使用contrast=number单独查看</span><br><span class="line">tamoxifen</span><br><span class="line">#按照分组分别进行差异分析，默认使用DESeq2进行计算，可以选择method = DBA_EDGER(edgR)，或者两个都要method = DBA_ALL_METHODS</span><br><span class="line">tamoxifen &lt;- dba.analyze(tamoxifen)</span><br><span class="line">dba.show(tamoxifen, bContrasts=TRUE)</span><br><span class="line">#查看差异分析的结果与导出为csv文件</span><br><span class="line">#查看第1组差异分析的结果</span><br><span class="line">tamoxifen.DB_1 &lt;- dba.report(tamoxifen,contrast=1)</span><br><span class="line">tamoxifen.DB_1</span><br><span class="line">write.csv(tamoxifen.DB_1, file=&quot;diffBind_result.csv&quot;)</span><br><span class="line"> </span><br><span class="line">#查看第2组差异分析的结果</span><br><span class="line">tamoxifen.DB_2 &lt;- dba.report(tamoxifen,contrast=2) tamoxifen.DB_2 write.csv(tamoxifen.DB_2, file=&quot;diffBind_result.csv&quot;) ##传统的上下调在找差异peak中称为“Gain”或“Loss” #查看第1组的差异peak数量“Fold&gt;0或&lt;0”控制是“Gain”或“Loss” sum(tamoxifen.DB$Fold&gt;0,contrast=1)</span><br><span class="line">sum(tamoxifen.DB$Fold&lt;0,contrast=1) #查看第2组的差异peak数量“Fold&gt;0或&lt;0”控制是“Gain”或“Loss” sum(tamoxifen.DB$Fold&gt;0,contrast=2)</span><br><span class="line">sum(tamoxifen.DB$Fold&lt;0,contrast=2)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<ul>
<li>由于我是使用两组进行测试的，所以可以使用 contrast=1 或 contrast=2 分别查看，由于都是相同的代码，下面进行可视化的时候，<strong>只选择 contrast=1 进行可视化</strong></li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">#韦恩图可视化</span><br><span class="line">dba.plotVenn(tamoxifen, contrast=1, bDB=TRUE, bGain=TRUE, bLoss=TRUE, bAll=FALSE)</span><br><span class="line"> </span><br><span class="line">#PCA</span><br><span class="line">dba.plotPCA(tamoxifen) #这里可以使用所有样本进行PCA</span><br><span class="line">dba.plotPCA(tamoxifen, contrast=1, label=DBA_FACTOR)#单独对分组1进行PCA</span><br><span class="line"> </span><br><span class="line">#MA plots</span><br><span class="line">dba.plotMA(tamoxifen, contrast=1)</span><br><span class="line"> </span><br><span class="line">#Volcano plots</span><br><span class="line">dba.plotVolcano(tamoxifen, contrast=1)</span><br><span class="line"> </span><br><span class="line">#Box plots</span><br><span class="line">dba.plotBox(tamoxifen, contrast=1)</span><br><span class="line"> </span><br><span class="line">#Heatmap</span><br><span class="line">hmap &lt;- colorRampPalette(c(&quot;red&quot;, &quot;black&quot;, &quot;green&quot;))(n = 13)</span><br><span class="line">#对所有样本的所有的差异peak画热图</span><br><span class="line">dba.plotHeatmap(tamoxifen,correlations=FALSE,scale=&quot;row&quot;,colScheme = hmap)</span><br><span class="line">#单独对分组1中所有的差异peak画热图</span><br><span class="line">dba.plotHeatmap(tamoxifen, contrast=1, correlations=FALSE, scale=&quot;row&quot;, colScheme = hmap)</span><br><span class="line">dba.plotHeatmap(tamoxifen,correlations=FALSE,scale=&quot;row&quot;,colScheme = hmap)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950027-1-700x349.png" alt=""></p>
<ul>
<li>想画下面的图，可是一直提示 could not find function dba.plotProfile，因此还没有成功，用官方提供的图展示一下吧，大家能画出来可以和我说一声哦。这是<a target="_blank" rel="noopener" href="https://content.cruk.cam.ac.uk/bioinformatics/software/DiffBind/plotProfileDemo.html">【官方文档】</a>可以参考</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">#Profiling and Profile Heatmaps</span><br><span class="line">dba.plotProfile(tamoxifen)</span><br><span class="line"></span><br></pre></td></tr></table></figure>
<p><img data-src="https://static.plob.org/wp-content/uploads/2021/09/1632950029-700x643.png" alt=""></p>
<h3 id="8-footprint-分析"><a href="#8-footprint-分析" class="headerlink" title="8. footprint 分析"></a>8. footprint 分析</h3><p>目前我还没有找到使用于所有物种的软件，HINT-ATAC 用来做 footprint 分析的话，应该只支持不几个物种，比较鸡肋。<a target="_blank" rel="noopener" href="https://bioconductor.org/packages/release/bioc/manuals/ATACseqQC/man/ATACseqQC.pdf">ATACseqQC</a> 应该是一款不错的软件，最近还在探索</p>
<h3 id="三、ATAC-seq-与多组学数据联合分析"><a href="#三、ATAC-seq-与多组学数据联合分析" class="headerlink" title="三、ATAC-seq 与多组学数据联合分析"></a>三、ATAC-seq 与多组学数据联合分析</h3><p><strong>转录因子 ChIP-seq</strong>：由于大部分转录因子结合的是染色质开放区域，所以 ATAC-seq 的 Peak 可能和转录因子 ChIP-seq 的 Peak 存在部分重叠的情况，而且 ATAC-seq 得到的 Peak 长度往往更长，因此 ATAC-seq 数据和转录因子 ChIP-seq 数据可以相互验证。转录因子在 ChIP-seq 中独有的 Peak 暗示这个转录因子可能是结合在异染色质区域的驱动型转录因子（Pioneer TFs），驱动型转录因子随后招募染色质重塑复合体以及其它转录因子开始转录。另外，联合分析已经报道的 ChIP-seq 数据可以更准确地分析转录因子的足迹。<br><strong>组蛋白修饰 ChIP-seq</strong>：ATAC-seq 数据同样可以和组蛋白修饰 ChIP-seq 数据进行联合分析，其中转录激活性修饰（H3K4me3，H3K4me1 和 H3K27ac 等）与染色质开放程度呈正相关，转录抑制性修饰（H3K27me3）与染色质开放程度呈负相关。联合已知的增强子和启动子之间的相互作用数据也可以帮助构建调控网络。<br><strong>RNA-seq</strong>：ATAC-seq 数据可以通过联合分析 RNA-seq 数据来发现哪些差异表达的基因是受染色质可及性调控的，进一步可以推测这些差异表达的基因哪些是受开放染色质中具有 motif 和 footprint 的转录因子调控的，因此 ATAC-seq 与 RNA-seq 的联合分析有助于破译基因调控网络和细胞异质性。</p>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="reward-container">
  <div>请我一杯咖啡吧！</div>
  <button>
    赞赏
  </button>
  <div class="post-reward">
      <div>
        <img src="/images/wechatpay.jpg" alt="Huang 微信">
        <span>微信</span>
      </div>
      <div>
        <img src="/images/alipay.jpg" alt="Huang 支付宝">
        <span>支付宝</span>
      </div>

  </div>
</div>

          

<div class="post-copyright">
<ul>
  <li class="post-copyright-author">
      <strong>本文作者： </strong>Huang
  </li>
  <li class="post-copyright-link">
      <strong>本文链接：</strong>
      <a href="https://yetu.fun/post/786e1faa.html" title="ATAC分析流程">https://yetu.fun/post/786e1faa.html</a>
  </li>
  <li class="post-copyright-license">
      <strong>版权声明： </strong>本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> 许可协议。转载请注明出处！
  </li>
</ul>
</div>

          <div class="post-tags">
              <a href="/tags/note/" rel="tag"><i class="fa fa-tag"></i> note</a>
              <a href="/tags/ATAC/" rel="tag"><i class="fa fa-tag"></i> ATAC</a>
              <a href="/tags/pipeline/" rel="tag"><i class="fa fa-tag"></i> pipeline</a>
          </div>
          <script type="text/javascript">
            var tagsall=document.getElementsByClassName("post-tags")
            for (var i = tagsall.length - 1; i >= 0; i--){
              var tags=tagsall[i].getElementsByTagName("a");
              for (var j = tags.length - 1; j >= 0; j--) {
                var golden_ratio = 0.618033988749895;
                var s = 0.5;
                var v = 0.999;
                var h = golden_ratio + Math.random()*0.8 - 0.5;
                var h_i = parseInt(h * 6);
                var f = h * 6 - h_i;
                var p = v * (1 - s);
                var q = v * (1 - f * s);
                var t = v * (1 - (1 - f) * s);
                var r, g, b;
                switch (h_i) {
                  case 0:
                    r = v;
                    g = t;
                    b = p;
                    break;
                  case 1:
                    r = q;
                    g = v;
                    b = p;
                    break;
                  case 2:
                    r = p;
                    g = v;
                    b = t;
                    break;
                  case 3 :
                    r = p;
                    g = q;
                    b = v;
                    break;
                  case 4:
                    r = t;
                    g = p;
                    b = v;
                    break;
                  case 5:
                    r = v;
                    g = p;
                    b = q;
                    break;
                  default:
                    r = 1;
                    g = 1;
                    b = 1;
                }
              tags[j].style.background = "rgba("+parseInt(r*255)+","+parseInt(g*255)+","+parseInt(b*255)+","+0.5+")";
              }
            }
          </script>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/post/b40e0536.html" rel="prev" title="免疫基础知识">
                  <i class="fa fa-chevron-left"></i> 免疫基础知识
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/post/fd2daa16.html" rel="next" title="Tfh Summary">
                  Tfh Summary <i class="fa fa-chevron-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>






</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">


<div class="copyright">
  &copy; 2022 – 
  <span itemprop="copyrightYear">2025</span>
  <span class="with-love">
    <i class="fa-solid fa-paw"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Copyright © Huang All Rights Reserved.</span>
</div>
<div class="wordcount">
  <span class="post-meta-item">
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-line"></i>
    </span>
    <span title="站点总字数">26k</span>
  </span>
  <span class="post-meta-item">
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="站点阅读时长">1:35</span>
  </span>
</div>
<div class="busuanzi-count">
    <span class="post-meta-item" id="busuanzi_container_site_uv">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="总访客量">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-item" id="busuanzi_container_site_pv">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="总访问量">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>

<!--
-->
    </div>
  </footer>

  

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/animejs/3.2.1/anime.min.js" integrity="sha256-XL2inqUJaslATFnHdJOi9GfQ60on8Wx1C2H8DYiN1xY=" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/lozad.js/1.16.0/lozad.min.js" integrity="sha256-mOFREFhqmHeQbXpK2lp4nA3qooVgACfh88fpJftLBbc=" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/pangu/4.0.7/pangu.min.js" integrity="sha256-j+yj56cdEY2CwkVtGyz18fNybFGpMGJ8JxG3GSyO2+I=" crossorigin="anonymous"></script>
<script src="/js/comments.js"></script><script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/next-boot.js"></script>

  <script src="https://cdnjs.cloudflare.com/ajax/libs/hexo-generator-searchdb/1.4.1/search.js" integrity="sha256-1kfA5uHPf65M5cphT2dvymhkuyHPQp5A53EGZOnOLmc=" crossorigin="anonymous"></script>
<script src="/js/third-party/search/local-search.js"></script>






  
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>




  

  <script class="next-config" data-name="enableMath" type="application/json">true</script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.7/katex.min.css" integrity="sha256-hLTCMFlKxdNgPXyWlSSxYN0ykJmxxq9Yt3MNfdRGWeA=" crossorigin="anonymous">



</body>
</html>
